### PR TITLE
fix: Remove "report problem" in the navigation bar if knowledge panel is not present

### LINKED ISSUE #10359: Remove "report a problem" entry in product navigation for producers platform
### Problem

On producer platform, we display the "report a problem" entry in product nav bar, while there is no corresponding knowledge panel, as it makes no sense (this is not public data).

### Proposed solution

Remove the entry in the menu if the knowledge panel is not there.

### PR DESCRIPTION
**Pull request by subi9**
"Report problem" navigation bar is removed if the corresponding knowledge panel is not present.Screenshot attached

### Screenshot
Screenshot from 2024-06-23 16-16-02

- Fixes #10359

### CODE REVIEW

**Review by stephanegigandet — APPROVED** [review]
Looks good, thank you!

**Comment by stephanegigandet** [comment]
Hi @subi9 , some of the tests results changed (just some new lines), could you run "make update_tests_results" to update the test results and commit the new results?

**Comment by subi9** [comment]
Hi @stephanegigandet , I have executed the "make update_tests_results" and then committed the results.Please approve if this looks good.

File: tests/integration/expected_test_results/api_v2_product_read/get-auth-bad-user-password.html [hunk 1]
```diff
@@ -1,7 +1,7 @@
 <!-- start templates/web/common/site_layout.tt.html -->
 
 <!doctype html>
-<html class="no-js" lang="en" data-serverdomain="openfoodfacts.localhost" dir="ltr">
+<html class="no-js" lang="en" data-serverdomain="pro.openfoodfacts.localhost" dir="ltr">
```
[the remark this answers is not in the document]
Comment (subi9) [reply tests/integration/expected_test_results/api_v2_product_read/get-auth-bad-user-password.html:4]: @stephanegigandet I have re executed the test results in the normal "off" environment. Can you please check and approve.

**Pull request merged**